### ISSUE TITLE
Bug: sub menus indexing

### ISSUE THREAD

**Issue by RyanPWalker** [nejcm/docz-theme-extended#2]
The new sub menu feature works as expected for my Accordion submenu.  However, my other two sub menus Forms and Pagination appear at the very bottom of the menu instead of alphabetically.  I've implemented them the same way so I don't see why they should appear at the bottom of the menu.

Accordion.mdx
```
---
menu: Components
name: Accordion
route: /Components/Accordion/Accordion
submenu: Accordion
---
```

Checkbox.mdx
```
---
menu: Components
name: Checkbox
route: /Components/Forms/Checkbox
submenu: Forms
---
```

![image](https://user-images.githubusercontent.com/25303720/87075294-58cd3c80-c1dd-11ea-8cb5-c1f33af3d28c.png)
![image](https://user-images.githubusercontent.com/25303720/87075325-61be0e00-c1dd-11ea-86c7-c0836d7475f9.png)

Let me know if you need any other info and if the screenshots don't show up.

**Comment by nejcm** [comment]
No other info needed. Currently there is no other ordering applied on the sub menus except the one docz offers with the menu option inside doczrc.js (https://www.docz.site/docs/project-configuration).
So sub menus are constructed based on the items that are received from that order.

I would suggest you try that first by grouping the items under same sub menu together at the top of the array or where you want them... something along the lines:
`
menu: [
    'Colors',  
    {  
       name: 'Components',  
       menu: [  
         'Accordion sub item #1',  
         'Accordion sub item #2',  
         'Pagination sub item #1',  
         'Pagination sub item #2',  
         'Alert',  
         'Button',  
      ]  
    }  
  ]  
`

In the example above this would order the items inside the Components menu as follows: Accordion sub menu, Pagination sub menu, Alert, Button. 

This is not tested so let me know if this worked or if we need to find a different solution.

I am aware this is not the cleanest solution to the problem... looks like more of a hack... but this is the easiest solution without completely changing some of the core mechanics of the base docz theme.

**Comment by RyanPWalker** [comment]
That did work and definitely looks better than before, thanks.  

![image](https://user-images.githubusercontent.com/25303720/87089031-026af880-c1f3-11ea-9d56-0c4457a95e71.png)

One thing to note is that it worked when I only specified the submenu components in the menu array.  If I tried to put other non submenu components in the array then it didn't work and would go back to how it was before.  I discovered this while trying to put all the components in the menu array in the order I wanted.

If you ever get around to a complete solution, let me know.  I think it's still worth looking in to if it doesn't tamper with the core mechanics of docz _too_ much.  But I get that it might be a lot of work.
